src/app/api/services/dot-interceptor/dot-interceptor.service.ts

```diff
@@ -0,0 +1,87 @@
+import { DotMessageService } from './../dot-messages-service';
+import { Injectable } from '@angular/core';
+import { RequestOptionsArgs, Response } from '@angular/http';
+
+import { Observable } from 'rxjs/Observable';
+
+import { ResponseView, CoreWebService, LoginService } from 'dotcms-js/dotcms-js';
+
+import { DotDialogService } from '../dot-dialog';
+import { DotRouterService } from '../dot-router-service';
+
+@Injectable()
+export class DotInterceptor {
+    constructor(
+        private coreWebService: CoreWebService,
+        private dotDialogService: DotDialogService,
+        private dotMessageService: DotMessageService,
+        private dotRouterService: DotRouterService,
+        private loginService: LoginService
+    ) {
+        this.dotMessageService.getMessages([
+            'dot.common.http.error.403.header',
+            'dot.common.http.error.403.message',
+            'dot.common.http.error.500.header',
+            'dot.common.http.error.500.message'
+        ]).subscribe();
+    }
+
+    /**
+     * Do request, show errors in the ui and do redirects
+     *
+     * @param {RequestOptionsArgs} options
+     * @returns {Observable<ResponseView>}
+     * @memberof DotInterceptor
+     */
+    request(options: RequestOptionsArgs): Observable<ResponseView> {
+        return this.coreWebService
+            .requestView(options)
+            .catch((err: ResponseView) => this.handleErrors(err))
+            .filter((res: ResponseView) => res !== null);
+    }
+
+    private handleErrors(err: ResponseView): Observable<ResponseView> {
+        this.callErrorHandler(err.response.status);
+        return Observable.of(null);
+    }
+
+    private callErrorHandler(code: number): void {
+        const errors = {
+            401: () => {
+                this.handle401();
+            },
+            403: () => {
+                this.handle403();
+            },
+            500: () => {
+                this.handle500();
+            }
+        };
```

Comment:
errors const don't must be define here, it must be define like a class attribute to be create just one time,also it should be static

Reply:
I found this

```
export declare enum HttpCode {
    BAD_REQUEST = 400,
    FORBIDDEN = 403,
    NOT_FOUND = 404,
    NO_CONTENT = 204,
    SERVER_ERROR = 500,
    UNAUTHORIZED = 401,
}
```

in dotcms-js so I use that... also we shouldn't use static because the last time I read (a while ago) it wasn't compatible with AoT.